src/cmap/conn/command.rs

```diff
@@ -120,11 +131,21 @@ impl CommandResponse {
         let cluster_time = raw_response
             .get("$clusterTime")
             .and_then(|subdoc| bson::from_bson(subdoc.clone()).ok());
+        let snapshot_time = raw_response
```

Comment:
My initial approach was to add `atClusterTime` as an explicit field in the output of the relevant commands (find, aggregate, distinct), but that required:
* introducing a trait for the output type of an operation, which added a lot of churn, and
* additional special handling in operations implemented on top of one of those three (e.g. countDocuments).

The approach here avoids both of those issues at the cost of digging a bit into the raw document.

Reply:
As part of #389, I actually did have to introduce a trait for the output type of an operation (there was definitely a lot of churn, but for that work it was required I think), so we could still go with your original plan once that PR is merged. For the purposes of avoiding merge conflicts, I think it'll probably be easier to merge this PR first though and then rebase #389 off of that and adjust for `atClusterTime` accordingly there.

Reply:
Ah, with that trait existing it definitely makes sense to go back to extracting the snapshot time that way.  Your plan for merging SGTM.